Troubleshooting: add A Request Error Occurs After Filtering or Search… (#5623)
DevExpress/devextreme-documentation: concepts/80 Troubleshooting/05 DataGrid and TreeList Issues/15 A Request Error Occurs After Filtering or Searching.md

> This topic addresses multiple request errors that can occur in your application. To check an error code, perform the following steps:

Reviewer: "may occur" is more suitable than "can occur" in everything regarding errors, for it's much softer and implies probability.

```suggestion
This topic addresses multiple request errors that may occur in your application. To check an error code, follow the steps below:
```

> 1. Obtain a URL link generated by the problematic data query in the browser [network tab](/Documentation/Guide/Troubleshooting/A_Drop-Down_Editor_Does_Not_Show_Data/#AspNetDatacreateStore_Method_Specifics). Paste the obtained URL into a new browser tab.

Reviewer 2: ```suggestion
1. Obtain a URL link generated by the problematic data query in the browser [network tab](/Documentation/Guide/Troubleshooting/A_Drop-Down_Editor_Does_Not_Show_Data/#AspNetDatacreateStore_Method_Specifics). Paste the URL in a new browser tab.
```

> 1. Obtain a URL link generated by the problematic data query in the browser's [network tab](/Documentation/Guide/Troubleshooting/A_Drop-Down_Editor_Does_Not_Show_Data/#AspNetDatacreateStore_Method_Specifics). Paste the obtained URL into a new browser tab.

Reviewer: In this scenario, "in the browser's network tab" means actually the same as "in the browser network tab". The latter is simpler, therefore, we should stick to a simpler option if it's possible.

```suggestion
1. Obtain a URL link generated by the problematic data query in the browser [network tab](/Documentation/Guide/Troubleshooting/A_Drop-Down_Editor_Does_Not_Show_Data/#AspNetDatacreateStore_Method_Specifics). Paste the obtained URL into a new browser tab.
```

> 2. Enable exceptions on the server to see the detailed error message. In ASP.NET, you can obtain an exception's call stack. Refer to the following help topic for more information: [How to obtain the exception's call stack](https://docs.devexpress.com/GeneralInformation/403685/support-debug-troubleshooting/obtain-exception-call-stack).

Reviewer: > Enable exceptions on the server to see **the** detailed error message.

We know nothing of this message just yet. Therefore, we should avoid the definite article case. Of course, you may have implied "**the** detailed **current** error message", however, it's not that important in this context.

Second.

> you can obtain an exception's call stack ...

Does the following form change your initial idea?

> In ASP.NET, you can obtain an exception call stack

There is no real need for using this genitive form. And, if we can rely on simpler forms -- we should.

Third:

>  Refer to the following help topic for more information: [How to obtain the exception's call stack](https://docs.devexpress.com/GeneralInformation/403685/support-debug-troubleshooting/obtain-exception-call-stack).

The original topic is named "Obtain an Exception's Call Stack", am I right?

https://docs.devexpress.com/GeneralInformation/403685/support-debug-troubleshooting/obtain-exception-call-stack

And you've changed the article case to definite for some reason. Since you already changed it, why don't you rectify it's original name and write the following:

> How to obtain an exception call stack

?

```suggestion
2. Enable exceptions on the server to see a detailed error message. In ASP.NET, you can obtain an exception call stack. Refer to the following help topic for more information: [How to obtain an exception call stack](https://docs.devexpress.com/GeneralInformation/403685/support-debug-troubleshooting/obtain-exception-call-stack).
```

> [DevExtreme.AspNet.Data](https://github.com/DevExpress/DevExtreme.AspNet.Data#devextreme-aspnet-data) sends load parameters (skip, take, filter, group) in the browser URL. As a result, an error may occur if a string with parameters is too long and cannot be processed on the server. Refer to the following article for more information: [Request Limits](https://docs.microsoft.com/en-us/iis/configuration/system.webserver/security/requestfiltering/requestlimits/). 

Reviewer: > (skip, take, filter, group)

From what I gather -- this is a full list.

```suggestion
[DevExtreme.AspNet.Data](https://github.com/DevExpress/DevExtreme.AspNet.Data#devextreme-aspnet-data) sends load parameters (skip, take, filter, and group) in the browser URL. As a result, an error may occur if a string with parameters is too long and cannot be processed on the server. Refer to the following article for more information: [Request Limits](https://docs.microsoft.com/en-us/iis/configuration/system.webserver/security/requestfiltering/requestlimits/). 
```

> The following errors can appear:

Reviewer: ```suggestion
The following errors may occur:
```

- `Query string is too long`
- `Request URI is too long`
- `Maximum request length exceeded`
- `404`
- `404.15`

> - The `object XMLHttpRequest` error occurs after adding in total summaries and group summaries.

Reviewer: Is this a standard error message? I'm not sure I understand this:

> ... after adding in total summaries and group summaries.

Adding what and adding where?

Author: I feel like I wanted to be more neutral here, but maybe the following would be better:

> The `object XMLHttpRequest` error occurs after you define total and group summaries in DataGrid.

Reviewer: OK, I see. In this case...

```suggestion
- The `object XMLHttpRequest` error occurs once you define total and group summaries in DataGrid.
```

- Filtering by `Search Panel` does not work if the component has a lookup column.
- Filtering by `Search Panel` does not work if the component has more than 20 columns.
- Selection does not work if [selection.deferred](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/selection/#deferred) is `true`.

> If you receive one of the errors above, consider the following solutions:

Reviewer: ```suggestion
If you receive one of the errors above, one of the following solutions may help you resolve it:
```

> - Increase the maximum query string length in your backend config.     

Reviewer: >  Increase the maximum query string length in your backend **config**.

"Config" sounds like slang here, which is not good for DevExpress documentation. Did you mean "in your backend configuration file"?

```suggestion
- Increase the maximum query string length in your backend configuration file.
```

> Note that browsers also have their own restrictions regarding the URL length: [What is the maximum length of a URL in different browsers?](https://stackoverflow.com/questions/417142/what-is-the-maximum-length-of-a-url-in-different-browsers).

Reviewer: ```suggestion
Note that browsers also impose their own restrictions on the maximum URL length: [What is the maximum length of a URL in different browsers?](https://stackoverflow.com/questions/417142/what-is-the-maximum-length-of-a-url-in-different-browsers).
```

            dataSource: DevExpress.data.AspNet.createStore({
                key: "keyFieldName",
                loadUrl: url,
                loadMethod: "POST",
            }),
            // ...
        });

    ##### Angular

        <!-- tab: app.component.ts -->
        import { createStore } from 'devextreme-aspnet-data-nojquery';

        const dataSource = createStore({
            key: "keyFieldName",
            loadUrl: urlHere,
            loadMethod: "Post",
            // ...
        });

    ##### Vue

        <!-- tab: App.vue -->
        <template>
            <!-- ... -->
        </template>

        <script>
        import { createStore } from 'devextreme-aspnet-data-nojquery';

        const dataSource = createStore({
            key: "keyFieldName",
            loadUrl: urlHere,
            loadMethod: "Post",
            // ...
        });
        </script>

    ##### React

        <!-- tab: App.js -->
        import { createStore } from 'devextreme-aspnet-data-nojquery';

        const dataSource = createStore({
            key: "keyFieldName",
            loadUrl: urlHere,
            loadMethod: "Post",
            // ...
        });

    ---

>     As `POST` request size becomes large after these changes, you may need to increase the `Content-Length` in the server config.

Reviewer: Again, "config" looks like slang to me.

Did you mean the following?

```suggestion
    As `POST` request size becomes large after these changes, you may need to increase the `Content-Length` value in the server configuration file.
```
